src/Infrastructure.EntityFramework/EntityFrameworkCache.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Bit.Infrastructure.EntityFramework.Repositories;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Internal;
+
+namespace Bit.Infrastructure.EntityFramework;
+
+public class EntityFrameworkCache : IDistributedCache
+{
+    private readonly DatabaseContext _databaseContext;
+    private readonly ISystemClock _systemClock;
+    private readonly TimeSpan _expiredItemsDeletionInterval = TimeSpan.FromMinutes(30);
+    private DateTimeOffset _lastExpirationScan;
+    private readonly Action _deleteExpiredCachedItemsDelegate;
+    private readonly TimeSpan _defaultSlidingExpiration = TimeSpan.FromMinutes(20);
+    private readonly object _mutex = new();
+
+    public EntityFrameworkCache(DatabaseContext databaseContext)
+    {
+        _databaseContext = databaseContext;
+        _systemClock = new SystemClock();
+        _deleteExpiredCachedItemsDelegate = DeleteExpiredCacheItems;
+    }
+
+    public byte[] Get(string key)
+    {
+        ArgumentNullException.ThrowIfNull(key);
+        var cache = _databaseContext.Cache
+            .Where(c => c.Id == key && c.ExpiresAtTime >= _systemClock.UtcNow)
+            .SingleOrDefault();
+        ScanForExpiredItemsIfRequired();
```

Comment:
💭 Strange placement to me. I guess this has its protections in place to not run too often but I wonder where it can run, and more async perhaps?

Reply:
Yes, see implementation below. It runs as a Task only every 30 minutes (`_expiredItemsDeletionInterval`). This is mostly copying https://github.com/dotnet/aspnetcore/blob/main/src/Caching/SqlServer/src/SqlServerCache.cs